engine/transaction/record/testabilities/assert_record_outline.go

```diff
@@ -0,0 +1,79 @@
+package testabilities
+
+import (
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/stretchr/testify/require"
+)
+
+type ErrorAssert interface {
+	NothingChanged()
+}
+
+type RecordOutlineAssert interface {
+	WithNoError(err error) SuccessfullyCreatedRecordOutlineAssertion
+	WithErrorIs(err, expectedError error) ErrorAssert
+
+	StoredOutputs([]database.Output) RecordOutlineAssert
```

Comment:
**Just a thought, don't have example 😅**
What do you think about checking if the final "repository" state matches exactly what we intended without any additional entries?

Reply:
I mean something like:
```go
    then.WithNoError(err).
        Broadcasted(givenTXWithOpReturn(t).ID()).
        StoredOutputs(expectedOutputs).
        StoredData(expectedData)
```

    actualOutputs := given.repository.GetAllOutputs()
    actualData := given.repository.GetAllData()

```go
    require.ElementsMatch(t, expectedOutputs, actualOutputs, "Repository contains unexpected outputs")
    require.ElementsMatch(t, expectedData, actualData, "Repository contains unexpected data")
```

And at the top define some expectedData used here as a parameter for require 😃

Reply:
I've changed from `Subset` to `ElementsMatch`, so now the behawior is as you described.

Reply:
Done 😄